contracts/AdvancedProviders/MultiWithdrawProvider/MultiWithdrawProvider.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./MultiWithdrawState.sol";
+
+contract MultiWithdrawProvider is MultiWithdrawState {
+    constructor(ILockDealNFT nftContract, uint256 _maxPoolsPerTx) {
+        name = "MultiWithdrawProvider";
+        lockDealNFT = nftContract;
+        maxPoolsPerTx = _maxPoolsPerTx;
+    }
+
+    function multiWithdrawAllPoolsOfOwner(address _owner) external {
+        uint256[] memory poolIds = getAllPoolsOfOwner(_owner);
+        _processMultiWithdraw(poolIds, _owner);
+    }
+
+    function multiWithdrawPoolsOfOwnerByVault(address _owner, uint256 _vaultId) external {
+        uint256[] memory poolIds = getPoolsOfOwnerByVault(_owner, _vaultId);
+        _processMultiWithdraw(poolIds, _owner);
+    }
+
+    function multiWithdrawPools(uint256[] memory _poolIds) external {
+        _processMultiWithdraw(_poolIds);
+    }
```

Comment:
need to add a check for the owner of the id. 
in the current state, any user can withdraw for any user, (like n V2) we want to keep the control to the user (in case of refund)